src/rewrite.rs

```diff
@@ -41,6 +42,7 @@ pub(crate) struct RewriteContext<'a> {
     pub(crate) macro_rewrite_failure: RefCell<bool>,
     pub(crate) report: FormatReport,
     pub(crate) skip_context: SkipContext,
+    pub(crate) skipped_range: Rc<RefCell<Vec<(usize, usize)>>>,
```

Comment:
This is an important point in this PR. I want to use skipped_range in RewriteContext. It's used to resolve #3717. I think the ```skipped_range``` should include the failed format span or skipped span. This is related with #3761.